packages/dd-trace/src/lambda/handler.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+const { channel } = require('../../../datadog-instrumentations/src/helpers/instrument')
+const { ERROR_MESSAGE, ERROR_TYPE } = require('../constants')
+const { ImpendingTimeout } = require('./runtime/errors')
+
+const globalTracer = global._ddtrace
+const tracer = globalTracer._tracer
+const timeoutChannel = channel('apm:aws:lambda:timeout')
+let __lambdaTimeout
+
+/**
+ * Publishes to the `apm:aws:lambda:timeout` channel when
+ * the AWS Lambda run time is about to end.
+ * Also subscribes to the `apm:aws:lambda:timeout` channel to
+ * call `crashFlush` in order to crash unfinished spans before timeout.
+ *
+ * @param {*} context AWS Lambda context object.
+ */
+function checkTimeout (context) {
+  let remainingTimeInMillis = context.getRemainingTimeInMillis()
+  const apmFlushDeadline = parseInt(process.env.DD_APM_FLUSH_DEADLINE)
+  if (apmFlushDeadline && apmFlushDeadline <= remainingTimeInMillis) {
+    remainingTimeInMillis = apmFlushDeadline
+  }
+
+  __lambdaTimeout = setTimeout(() => {
+    timeoutChannel.publish(undefined)
+  }, remainingTimeInMillis - 50)
+
+  timeoutChannel.subscribe(_ => {
```

Comment:
It appears that `checkTimeout` is called for each function call? So shouldn't this be subscribed _outside_ of `checkTimeout` so it doesn't get attached repeatedly? The way it is here each new request would add another subscriber which would receive all future published events.

Reply:
Moved to the top-level of the file as suggested. This code was indeed over subscribing. 😖